### PR TITLE
Add search and Ballerina pull functionalities

### PR DESCRIPTION
**Pull request by IrushiL**
## Purpose
* Add the functionality to pull a Ballerina module template from Ballerina Central and use it for creating new modules.
* Add the search functionality to filter out templates from name in the Ballerina Integrator VS Code extension.
* When creating a new module from a template, a step is added to create a new project or choose an existing project.
* The project location folder picker is given more appropriate names.
* An add module step is added to the create new project component.

### CODE REVIEW

File: vscode/extension/src/createProject.ts [hunk 1]
```diff
@@ -34,69 +34,159 @@ export async function createTemplateProject(currentPanel: vscode.WebviewPanel, c
     let projectName = undefined;
     let folderPath = undefined;
     let projectPath = undefined;
+    let uri = undefined;
+    let projectUri = undefined;
     currentPanel.webview.onDidReceiveMessage(
         async homePageMessage => {
+            const cp = require('child_process');
```
Comment (dakshika) [inline vscode/extension/src/createProject.ts]: better to rename with proper const name

File: vscode/extension/src/createProject.ts [hunk 2]
```diff
@@ -34,69 +34,159 @@ export async function createTemplateProject(currentPanel: vscode.WebviewPanel, c
...
+                            resolve();
                         });
                     }
                 }
                 currentPanel.dispose();
-                vscode.commands.executeCommand("ballerinaIntegrator.projectTemplates");
+                vscode.commands.executeCommand("ballerina.integrator.activate");
             } else {
+                const addListCommand = "ballerina add --list";
+                const pullCommand = 'ballerina pull wso2/' + templateSelected;
```
Comment (dakshika) [inline vscode/extension/src/createProject.ts]: follow one pattern

File: vscode/extension/src/createProject.ts [hunk 3]
```diff
@@ -34,69 +34,159 @@ export async function createTemplateProject(currentPanel: vscode.WebviewPanel, c
...
-                                    window.showErrorMessage(stderr);
+                    // Get input on where the new module should be created.
+                    const projectOption = await vscode.window.showQuickPick(['New Project', 'Existing Project'],
+                        { placeHolder: 'Creating a new project or adding the module to an existing project?' });
+                    // If the module is to be created inside a new project.
+                    if (projectOption != undefined) {
+                        if (projectOption == "New Project") {
+                            // Get project name from the user.
+                            projectName = await window.showInputBox({
+                                prompt: "Enter value for project name"
```
Comment (dakshika) [inline vscode/extension/src/createProject.ts]: remove value from here

File: vscode/extension/src/createProject.ts (hunk 1 above)
Comment (IrushiL) [reply vscode/extension/src/createProject.ts]: Done

File: vscode/extension/src/createProject.ts (hunk 2 above)
Comment (IrushiL) [reply vscode/extension/src/createProject.ts]: Done

File: vscode/extension/src/createProject.ts (hunk 3 above)
Comment (IrushiL) [reply vscode/extension/src/createProject.ts]: Done

**Pull request merged**